### PR TITLE
[trino-main] Automatically refactor test code to use AssertJ

### PR DESCRIPTION
**Pull request by wendigo**
This was achieved using OpenRewrite + Refaster rules from https://github.com/PicnicSupermarket/error-prone-support/tree/master/error-prone-contrib/src/main/java/tech/picnic/errorprone/refasterrules

All tests are passing locally for me

The end goal is to remove the TestNG assertions and move code to JUnit 5

### CODE REVIEW

**Comment by wendigo** [comment]
I started with the toughest module first

File: core/trino-main/src/test/java/io/trino/sql/planner/iterative/rule/test/RuleTester.java [hunk 1]
```diff
@@ -74,7 +74,7 @@ public RuleTester(LocalQueryRunner queryRunner)
         this.typeAnalyzer = createTestingTypeAnalyzer(queryRunner.getPlannerContext());
     }
 
-    public RuleAssert assertThat(Rule<?> rule)
+    public RuleAssert assertRule(Rule<?> rule)
```
Comment (ksobolew) [inline core/trino-main/src/test/java/io/trino/sql/planner/iterative/rule/test/RuleTester.java:77]: It's an instance method - will it actually cause a name clash?

File: core/trino-main/src/test/java/io/trino/block/TestDictionaryBlock.java [hunk 2]
```diff
@@ -353,27 +349,15 @@ private static void assertDictionarySizeMethods(Block block)
 
         boolean[] selectedPositions = new boolean[positions];
         selectedPositions[0] = true;
-        assertEquals(
-                DictionaryBlock.create(allIds.length, block, allIds).getPositionsSizeInBytes(selectedPositions, 1),
-                block.getPositionsSizeInBytes(selectedPositions, 1) + Integer.BYTES);
+        assertThat(DictionaryBlock.create(allIds.length, block, allIds).getPositionsSizeInBytes(selectedPositions, 1)).isEqualTo(block.getPositionsSizeInBytes(selectedPositions, 1) + Integer.BYTES);
```
Comment (ksobolew) [inline core/trino-main/src/test/java/io/trino/block/TestDictionaryBlock.java:352]: Some manual post-processing to adds line wraps would be nice :)

Comment (wendigo) [reply core/trino-main/src/test/java/io/trino/block/TestDictionaryBlock.java:352]: Too much to handle I think.

File: pom.xml [hunk 3]
```diff
@@ -2343,6 +2346,17 @@
...
+                                    <groupId>tech.picnic.error-prone-support</groupId>
+                                    <artifactId>error-prone-contrib</artifactId>
+                                    <version>${dep.errorprone-support.version}</version>
+                                </path>
+                                <!-- Error Prone Support's Refaster rules. -->
+                                <path>
+                                    <groupId>tech.picnic.error-prone-support</groupId>
+                                    <artifactId>refaster-runner</artifactId>
+                                    <version>${dep.errorprone-support.version}</version>
+                                </path>
```
[the remark this answers is not in the document]
Comment (wendigo) [reply pom.xml]: Removed

File: core/trino-main/src/test/java/io/trino/sql/planner/iterative/rule/test/RuleTester.java (hunk 1 above)
Comment (wendigo) [reply core/trino-main/src/test/java/io/trino/sql/planner/iterative/rule/test/RuleTester.java:77]: yeah, static import + method in a class

**Comment by martint** [comment]
Instead of renaming those `assertThat` methods, they should be turned into `AssertProvider`, similar to how `io.trino.sql.query.QueryAssertions.query()` and friends work.

**Comment by martint** [comment]
I'm still reviewing, but so we don't forget, the last two commits need to be squashed before merging:
* Automatically refactor code to use AssertJ
* Manual fixes to automatically refactored code

**Pull request closed without merging**